scripts/tslint/rules/noDoubleSpaceRule.ts

```diff
@@ -38,17 +38,7 @@ function walk(ctx: Lint.WalkContext<void>): void {
 }
 
 function countDoubleSpaces(line: string): number {
-    let count = 0;
-    let idx = 0;
-    while (true) {
-        idx = line.indexOf("  ", idx);
-        if (idx === -1) {
-            return count;
-        }
-        // Start searching at the *next* character
-        idx++;
-        count++;
-    }
+    return (line.match(/\s\s/g) || []).length;

```

Comment:
This matches on `a    b` (4 spaces) twice. Is that intentional?

Reply:
The goal of this function is to ignore the rule on uses of multiple spaces that are probably not just accidentally pressing the space bar twice -- we could eliminate this entirely in a followup PR, but if we change to `/\s\s+/g` we would get some new errors right now.

Reply:
There are multiple ways we might want to make this rule stricter, such as not allowing aligning `=` outside of enums (or even within them).

Reply:
Is the heuristic here supposed to be the length of a run of whitespace (for marking that specific run as an error), or the number of individual whitespace runs on the line (for marking the all instances on the line as an error)? As is, it's half the total length of each >1 length run of whitespace on the line, which seems incorrect, since it's neither of those.

Reply:
I've fixed the remaining cases, so we don't need this function any more.